UnitySDK/Assets/ML-Agents/Editor/Tests/RandomNormalTest.cs

```diff
@@ -1,122 +1,163 @@
-		{
-			RandomNormal rn = new RandomNormal(1982);
-			Tensor t = new Tensor
-			{
-				ValueType = Tensor.TensorType.FloatingPoint,
-				Data = Array.CreateInstance(typeof(float), new long[3] {3, 4, 2})
-		};
-
-			rn.FillTensor(t);
-
-			float[] reference = new float[]
-			{
-				-0.2139822f,
-				0.5051259f,
-				-0.5640336f,
-				-0.3357787f,
-				-0.2055894f,
-				-0.09432302f,
-				-0.01419199f,
-				0.53621f,
-				-0.5507085f,
-				-0.2651141f,
-				0.09315512f,
-				-0.04918706f,
-				-0.179625f,
-				0.2280539f,
-				0.1883962f,
-				0.4047216f,
-				0.1704049f,
-				0.5050544f,
-				-0.3365685f,
-				0.3542781f,
-				0.5951571f,
-				0.03460682f,
-				-0.5537263f,
-				-0.4378373f,
-			};
-
-			int i = 0;
-			foreach (float f in t.Data)
-			{
-				Assert.AreEqual(f, reference[i], 0.0001);
-				++i;
-			}
-
-
-		}
-	}
+    public class RandomNormalTest
+    {
+        private const float first = -1.19580f;
+        private const float second = -0.97345f;
+
+        [Test]
+        public void RandomNormalTestTwoDouble ()
+        {
+            RandomNormal rn = new RandomNormal (2018);
+
+            Assert.AreEqual (first, rn.NextDouble (), 0.0001);
+            Assert.AreEqual (second, rn.NextDouble (), 0.0001);
+        }
+
+        [Test]
+        public void RandomNormalTestWithMean ()
+        {
+            RandomNormal rn = new RandomNormal (2018, 5.0f);
+
+            Assert.AreEqual (first + 5.0, rn.NextDouble (), 0.0001);
+            Assert.AreEqual (second + 5.0, rn.NextDouble (), 0.0001);
+        }
+
+        [Test]
+        public void RandomNormalTestWithStddev ()
+        {
+            RandomNormal rn = new RandomNormal (2018, 0.0f, 4.2f);
+
+            Assert.AreEqual (first * 4.2, rn.NextDouble (), 0.0001);
+            Assert.AreEqual (second * 4.2, rn.NextDouble (), 0.0001);
+        }
+
+        [Test]
+        public void RandomNormalTestWithMeanStddev ()
+        {
+            float mean = -3.2f;
+            float stddev = 2.2f;
+            RandomNormal rn = new RandomNormal (2018, mean, stddev);
+
+            Assert.AreEqual (first * stddev + mean, rn.NextDouble (), 0.0001);
+            Assert.AreEqual (second * stddev + mean, rn.NextDouble (), 0.0001);
+        }
+
+        [Test]
+        public void RandomNormalTestTensorInt ()
+        {
+            RandomNormal rn = new RandomNormal (1982);
+            Tensor t = new Tensor {
+                ValueType = Tensor.TensorType.Integer
+            };
+
+            Assert.Throws<NotImplementedException> (() => rn.FillTensor (t));
+        }
+
+        [Test]
+        public void RandomNormalTestDataNull ()
+        {
+            RandomNormal rn = new RandomNormal (1982);
+            Tensor t = new Tensor {
+                ValueType = Tensor.TensorType.FloatingPoint
+            };
+
+            Assert.Throws<ArgumentNullException> (() => rn.FillTensor (t));
+        }
+
+        [Test]
+        public void RandomNormalTestDistribution ()
```

Comment:
This test fails before the fix; the mean comes out right but the stddev is way off.